exasol/error/_parse.py

```diff
@@ -152,134 +155,162 @@ def validate(self, node: ast.Call, file: str) -> Result:
-                            type="mitigations", value=type(e)
-                        ),
-                        file=file,
-                        line_number=e.lineno,
-                    )
-                    for e in invalid
+
+    def _validate_mitigations(self, node: ast.expr, file: str) -> Optional[List[str]]:
+        if mitigation := self._check_node_types(
+            ast.List, ast.Constant, node, "mitigations", file
+        ):
+            if isinstance(mitigation, ast.List):
+                invalid = [
+                    e for e in mitigation.elts if not isinstance(e, ast.Constant)
                 ]
-            )
-            if invalid:
-                return None
-            else:
-                return [e.value for e in node.elts if isinstance(e, ast.Constant)]
-        elif isinstance(node, ast.Constant):
-            return [node.value]
+                self._errors.extend(
+                    [
+                        Error(code=INVALID_ERROR_CODE_DEFINITION.identifier,
+                              message=INVALID_ERROR_CODE_DEFINITION.message,
+                              mitigations=INVALID_ERROR_CODE_DEFINITION.mitigations,
+                              parameters={"error_element": "mitigations", "file": file, "line": node.lineno, "defined_type": type(e)})
+                        for e in invalid
+                    ]
+                )
+                if invalid:
+                    return None
+                else:
+                    return [
+                        e.value for e in mitigation.elts if isinstance(e, ast.Constant)
+                    ]
+            elif isinstance(mitigation, ast.Constant):
+                return [mitigation.value]
         return None
 
+    def normalize(self, params):
+        for k, v in zip(params.keys, params.keys):
+            if isinstance(v, ast.Call):
+                yield k.value, v[1]
+            else:
+                yield k.value, ""
+
+    def _validate_parameter_keys(self, parameter_node: ast.Dict, file: str):
+        ret_val = True
+        for key in parameter_node.keys:
+            if key is None:
+                self._errors.append(
+                    Error(code=INVALID_ERROR_CODE_DEFINITION.identifier,
+                          message=INVALID_ERROR_CODE_DEFINITION.message,
+                          mitigations=INVALID_ERROR_CODE_DEFINITION.mitigations,
+                          parameters={"error_element": "parameter keys", "file": file, "line": parameter_node.lineno,
+                                      "defined_type": "NoneType"})
+                )
+                ret_val = False
+            elif not self._check_node_type(
+                ast.Constant, key, "key", file
+            ):
+                ret_val = False
+        return ret_val
+
+    def _validate_parameter_values(
+        self, pparameter_node: ast.Dict, file: str
+    ) -> bool:
+        ret_val = True
```

Comment:
see https://github.com/exasol/error-reporting-python/pull/58/commits/aae099fb142fe76b8a41185bcaa61ebe1b77f45d#r1958293721

Reply:
Hm, not sure what you mean

Reply:
Fixed